src/hatti/ona/forms.cljs

```diff
@@ -279,32 +279,37 @@
     (cond-> field (not label) (assoc :label label))))
 
 ;; UTILITY: Form Flattening
+(defn name-label-map
+  [{:keys [children name label] :as child} prefix acc flatten-repeats?]
+  (let [full-name       (if prefix (str prefix "/" name) name)
+        langs           (when (map? label) (keys label))
+        updated-child   (with-meta
+                          (assoc child :full-name full-name) {:langs langs})
+        childless-child (dissoc updated-child :children)
+        recur'          (map #(name-label-map % full-name [] flatten-repeats?)
```

Comment:
I think the previous name is clearer, `new-children` but could be `make-new-children` I definitely don't like the mysterious single quote at the end